include/net/ieee802154_radio.h

```diff
@@ -54,14 +54,44 @@ struct ieee802154_filter {
 /* @endcond */
 };
 
+/** IEEE802.15.4 Transmission mode. */
+enum ieee802154_tx_mode {
+	/** Transmit packet immediately, no CCA. */
+	IEEE802154_TX_MODE_PLAIN,
+
+	/** Perform CCA before packet trasmission. */
+	IEEE802154_TX_MODE_CCA,
+
+	/** Perform full CSMA CA procedure before packet transmission. */
+	IEEE802154_TX_MODE_CSMA_CA,
+
+	/** Transmit packet in the future, at specified time, no CCA. */
+	IEEE802154_TX_MODE_TXTIME,
```

Comment:
don't we need additional capability for delaying tx?

Reply:
Good point, will add.